db/upgrade.php

```diff
@@ -230,6 +230,33 @@ function xmldb_bigbluebuttonbn_upgrade($oldversion = 0) {
         // Update db version tag.
         upgrade_mod_savepoint(true, 2019101001, 'bigbluebuttonbn');
     }
+    if ($oldversion < 2020072001) {
+        // Add field for guestlinkenabled.
+        $fielddefinition = array('type' => XMLDB_TYPE_INTEGER, 'precision' => '1', 'unsigned' => null,
+            'notnull' => XMLDB_NOTNULL, 'sequence' => null, 'default' => 0, 'previous' => null);
+        xmldb_bigbluebuttonbn_add_change_field($dbman, 'bigbluebuttonbn', 'guestlinkenabled', $fielddefinition);
+        // Add field for guestlinkid.
+        $fielddefinition = array('type' => XMLDB_TYPE_CHAR, 'precision' => '255', 'unsigned' => null,
+            'notnull' => null, 'sequence' => null, 'default' => null, 'previous' => null);
+        xmldb_bigbluebuttonbn_add_change_field($dbman, 'bigbluebuttonbn', 'guestlinkid', $fielddefinition);
+        // Add field for guestpass.
+        $fielddefinition = array('type' => XMLDB_TYPE_INTEGER, 'precision' => '10', 'unsigned' => null,
+            'notnull' => null, 'sequence' => null, 'default' => null, 'previous' => null);
+        xmldb_bigbluebuttonbn_add_change_field($dbman, 'bigbluebuttonbn', 'guestpass', $fielddefinition);
+        try {
```

Comment:
Still working on this review. This is brutal for massive databases. I am thinking that a better approach is to keep the empty and change the way they are implemented in the UI. Actually, base on this idea, the feature comes disabled by default which is what we want.

So, if the gueslinkid is empty, the feature is disabled. In the setting there should be only a checkbox that when ticked, it would open any other setting needed. On save, a new guestlinkid would be generated.

The UI only works is guestlink feature is enabled and properly set, so, also the view must be changed

Reply:
This is a good idea, I changed that in the updated version of the code.

Reply:
With the current behavior, the link changes on every save button press. So if you publish a link to your external auditors, you must not change conference presets any more. That's a big issue as the moderator is not warned about that. Just changing a typo in room description will invalidate all links.